db/migrate/20230823001215_create_mentee_application_states.rb

```diff
@@ -0,0 +1,12 @@
+class CreateMenteeApplicationStates < ActiveRecord::Migration[7.0]
+  def change
+    create_table :mentee_application_states do |t|
+      t.references :user_mentee_application, null: false, foreign_key: true
+      t.integer :status, null: false, default: 0
+      t.text :note
+      t.integer :status_changed_by_id
```

Comment:
this should be some type of t.references with like `foreign_key: { to_table: :users }` or something like that.